ibm/service/vpc/resource_ibm_is_instance_network_interface_floating_ip.go

```diff
@@ -0,0 +1,355 @@
+	"log"
+	"strings"
+	"time"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	isInstanceNetworkInterfaceFloatingIpAvailable = "available"
+	isInstanceNetworkInterfaceFloatingIpDeleting  = "deleting"
+	isInstanceNetworkInterfaceFloatingIpPending   = "pending"
+	isInstanceNetworkInterfaceFloatingIpDeleted   = "deleted"
+	isInstanceNetworkInterfaceFloatingIpFailed    = "failed"
+	isInstanceNetworkInterface                    = "network_interface"
+	isInstanceNetworkInterfaceFloatingIPID        = "floating_ip"
+)
+
+func ResourceIBMIsInstanceNetworkInterfaceFloatingIp() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMISInstanceNetworkInterfaceFloatingIpCreate,
+		ReadContext:   resourceIBMISInstanceNetworkInterfaceFloatingIpRead,
+		UpdateContext: resourceIBMISInstanceNetworkInterfaceFloatingIpUpdate,
+		DeleteContext: resourceIBMISInstanceNetworkInterfaceFloatingIpDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			isInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Instance identifier",
+			},
+			isInstanceNetworkInterface: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Instance network interface identifier",
+			},
+			isInstanceNetworkInterfaceFloatingIPID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The floating ip identifier of the network interface associated with the Instance",
+			},
+			floatingIPName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the floating IP",
+			},
+
+			floatingIPAddress: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Floating IP address",
+			},
+
+			floatingIPStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Floating IP status",
+			},
+
+			floatingIPZone: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Zone name",
+			},
+
+			floatingIPTarget: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Target info",
+			},
+
+			floatingIPCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Floating IP crn",
+			},
+		},
+	}
+}
+
+func resourceIBMISInstanceNetworkInterfaceFloatingIpCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	instanceId := ""
+	if instanceIdOk, ok := d.GetOk(isInstanceID); ok {
```

Comment:
Why GetOk instanceid,isInstanceNetworkInterface,instanceNicFipId all are required can we use d.Get()